src/app/components/UptimeStatus/index.tsx

```diff
@@ -1,61 +1,64 @@
 import { FC } from 'react'
-import Box from '@mui/material/Box'
-import { styled } from '@mui/material/styles'
+import { useTranslation } from 'react-i18next'
 import { COLORS } from '../../../styles/theme/colors'
 import { PercentageValue } from '../PercentageValue'
+import { cn } from '@oasisprotocol/ui-library/src/lib/utils'
+import { ValidatorUptime } from 'oasis-nexus/api'
 
 const getUptimeItemColor = (value: number | undefined) => {
-  if (!value) {
+  if (typeof value !== 'number') {
     return COLORS.grayMediumLight
   }
 
-  if (value > 90) {
+  if (value > 900) {
     return COLORS.eucalyptus
   }
 
   return COLORS.errorIndicatorBackground
 }
 
-type UptimeItem = {
-  small?: boolean
-  value?: number
-}
-
-const StyledBox = styled(Box)<UptimeItem>(({ small, value }) => ({
-  background: getUptimeItemColor(value),
-  width: small ? '3px' : '7px',
-  minWidth: small ? '3px' : '7px',
-  height: small ? '15px' : '45px',
-  borderRadius: small ? '2px' : '4px',
-  color: COLORS.white,
-  marginRight: small ? 1 : 2,
-}))
-
 export const ensureTwelveElements = (inputArray: number[] = []) => {
   return [...inputArray, ...new Array(12).fill(undefined)].slice(0, 12)
 }
 
 type UptimeStatusProps = {
-  percentage?: number
   small?: boolean
-  status?: number[]
+  uptime?: ValidatorUptime
 }
 
-export const UptimeStatus: FC<UptimeStatusProps> = ({ percentage, status, small }) => {
-  const adjustedStatus = ensureTwelveElements(status)
+export const UptimeStatus: FC<UptimeStatusProps> = ({ uptime, small }) => {
+  const { t } = useTranslation()
+
+  if (!uptime?.segment_uptimes) {
+    return <>{t('common.missing')}</>
+  }
+
+  const adjustedStatus = ensureTwelveElements(uptime?.segment_uptimes)
 
   return (
-    <Box sx={{ display: 'flex', alignItems: 'center', gap: small ? 4 : 5 }}>
-      <Box sx={{ display: 'flex' }}>
-        {adjustedStatus?.map((value, index) => (
-          <StyledBox small={small} key={`${value}-${index}`} value={value} />
-        ))}
-      </Box>
-      {percentage && (
-        <Box>
-          <PercentageValue value={percentage} />
-        </Box>
+    <div className={cn('flex items-center gap-4 justify-between', small && 'w-[110px]')}>
+      <div className="flex">
+        {adjustedStatus
+          ?.reverse()
+          .map((value, index) => (
+            <div
+              key={`${value}-${index}`}
+              className={cn(
+                'inline-block',
+                small
+                  ? 'w-[3px] min-w-[3px] h-[15px] rounded-[2px] mr-[1px]'
+                  : 'w-[7px] min-w-[7px] h-[45px] rounded-[4px] mr-[2px]',
+              )}
+              style={{ backgroundColor: getUptimeItemColor(value), color: COLORS.white }}
```

Comment:
Current example:
`[ 1200, 1054, 1037, 1047, 929, 1200, 939, 1055, 870, 1200, 947, 1198 ]` ==> 
<img width="311" height="136" alt="image" src="https://github.com/user-attachments/assets/40ea7025-5c40-471b-9383-eb1889344e75" />

If I understand this correctly, we only show red/green bars, and nothing else. The number goes 0-1200, and if we are above 999 it's green, if below, red. But there is no distinction between 100 and 900, or 1000 and 1200. Could we, for example, make the bar length proportionate to the actual number? (Keeping the color of course, which is a good first glance indication.)